src/gui/qgsabstractdatasourcewidget.cpp

```diff
@@ -58,9 +58,10 @@ void QgsAbstractDataSourceWidget::setupButtons( QDialogButtonBox *buttonBox )
   connect( closeButton, &QPushButton::clicked, this, &QgsAbstractDataSourceWidget::reject );
 }
 
-void QgsAbstractDataSourceWidget::setMapCanvas( const QgsMapCanvas *mapCanvas )
+void QgsAbstractDataSourceWidget::setMapCanvas( QgsMapCanvas *mapCanvas )
 {
   mMapCanvas = mapCanvas;
+  emit mapCanvasChanged();
```

Comment:
Instead of the signal, I'd make setMapCanvas virtual and override it in QgsOWSSourceSelect with the desired logic